File: 004mostFrequentChar.js
const mostFrequentChar = (s) => {
  // todo
  const map = {};
  let maxCount = 0;
  let maxChar;
  for (let char of s) {
    if (!(char in map)) {
      map[char] = 0;
    }
    map[char]++;
  }
  for (let char in map) {
    if (map[char] > maxCount) {
      maxCount = map[char];
      maxChar = char;
    }
  }
  return maxChar;
};


console.log(mostFrequentChar('bookeeper')); // -> 'e'
console.log(mostFrequentChar('david')); // -> 'd'
console.log(mostFrequentChar('abby')); // -> 'b'
console.log(mostFrequentChar('mississippi')); // -> 'i'

File: 004most_frequent_char.py
def most_frequent_char(s):
    count = {}
    for char in s:
        if char not in count:
            count[char] = 0

        count[char] += 1

    max_char = None
    for char in s:
      if max_char is None or count[char] > count[max_char]:
        max_char = char

    return max_char
